Updated hibernate to version 5.6.14

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4786

Updated hibernate to 5.6.14, Also removed deprecated TypeConfigurationRegistry class and added javassist dependency in pom because it was removed in Hibernate 5.6 and is not included in it

MarekB01 committed:
BroadleafCommerce/QA-4786_libraries_upgrade

Updated hibernate to 5.6.14, Also removed deprecated TypeConfigurationRegistry class and added javassist dependency in pom because it was removed in Hibernate 5.6 and is not included in it

oleksiimiroshnyk commented on common/pom.xml:
```diff
@@ -442,5 +442,10 @@
+        <dependency>
+            <groupId>org.javassist</groupId>
+            <artifactId>javassist</artifactId>
+            <version>3.22.0-GA</version>
```
i'm not sure if we want to specify version here.
I think we usually have somewhere in "root" poms dependencyManagement section where we declare dependencies and its versions... can you please check ?

MarekB01 committed:
BroadleafCommerce/QA-4786_libraries_upgrade

Removed version of javassist from common pom.xml and added it to parent pom. Made the same with the protobuf-java dependency
